src/se/element.rs

```diff
@@ -0,0 +1,1455 @@
+                            level: QuoteLevel::Full,
+                        },
+                        key: XmlName("root"),
+                    };
+
+                    match $data.serialize(ser).unwrap_err() {
+                        DeError::$kind(e) => assert_eq!(e, $reason),
+                        e => panic!(
+                            "Expected `{}({})`, found `{:?}`",
+                            stringify!($kind),
+                            $reason,
+                            e
+                        ),
+                    }
+                    // We can write something before fail
+                    // assert_eq!(buffer, "");
+                }
+            };
+        }
+
+        serialize_as!(false_: false => "<root>false</root>");
+        serialize_as!(true_:  true  => "<root>true</root>");
+
+        serialize_as!(i8_:    -42i8                => "<root>-42</root>");
+        serialize_as!(i16_:   -4200i16             => "<root>-4200</root>");
+        serialize_as!(i32_:   -42000000i32         => "<root>-42000000</root>");
+        serialize_as!(i64_:   -42000000000000i64   => "<root>-42000000000000</root>");
+        serialize_as!(isize_: -42000000000000isize => "<root>-42000000000000</root>");
+
+        serialize_as!(u8_:    42u8                => "<root>42</root>");
+        serialize_as!(u16_:   4200u16             => "<root>4200</root>");
+        serialize_as!(u32_:   42000000u32         => "<root>42000000</root>");
+        serialize_as!(u64_:   42000000000000u64   => "<root>42000000000000</root>");
+        serialize_as!(usize_: 42000000000000usize => "<root>42000000000000</root>");
+
+        serde_if_integer128! {
+            serialize_as!(i128_: -420000000000000000000000000000i128 => "<root>-420000000000000000000000000000</root>");
+            serialize_as!(u128_:  420000000000000000000000000000u128 => "<root>420000000000000000000000000000</root>");
+        }
+
+        serialize_as!(f32_: 4.2f32 => "<root>4.2</root>");
+        serialize_as!(f64_: 4.2f64 => "<root>4.2</root>");
+
+        serialize_as!(char_non_escaped: 'h' => "<root>h</root>");
+        serialize_as!(char_lt:   '<' => "<root>&lt;</root>");
+        serialize_as!(char_gt:   '>' => "<root>&gt;</root>");
+        serialize_as!(char_amp:  '&' => "<root>&amp;</root>");
+        serialize_as!(char_apos: '\'' => "<root>&apos;</root>");
+        serialize_as!(char_quot: '"' => "<root>&quot;</root>");
+
+        serialize_as!(str_non_escaped: "non-escaped string" => "<root>non-escaped string</root>");
+        serialize_as!(str_escaped: "<\"escaped & string'>" => "<root>&lt;&quot;escaped &amp; string&apos;&gt;</root>");
+
+        err!(bytes: Bytes(b"<\"escaped & bytes'>") => Unsupported("`serialize_bytes` not supported yet"));
+
+        serialize_as!(option_none: Option::<&str>::None => "<root/>");
+        serialize_as!(option_some: Some("non-escaped string") => "<root>non-escaped string</root>");
```

Comment:
Consider testing serialization of `Option(&str)` with escapable input

Reply:
And similar variants

Reply:
The point here is to test serialization of `Some` variant. Using a string just a most straightforward choice for a type. I think it is enough to test only the one variant of composition here, we should assume that others will work too, otherwise it is unclear where we should stop and believe that composition works